tests/drivers/pwm/pwm_api/src/test_pwm.c

```diff
@@ -159,4 +159,5 @@ ZTEST_USER(pwm_basic, test_pwm_cycle)
 	/* Period : Pulse (64000 : 0), unit (cycle). Voltage : 0V */
 	zassert_true(test_task(DEFAULT_PWM_PORT, DEFAULT_PERIOD_CYCLE,
 				0, UNIT_CYCLES) == TC_PASS, NULL);
+	k_sleep(K_MSEC(1000));
```

Comment:
I do not understand why this is needed. Please elaborate a bit.

Reply:
I think the issue here is that with the new ZTEST framework, the `test_pwm_cycle` test will run, and directly afterwards the `test_pwm_nsec` test will run. Without this delay, the 0% duty cycle signal will only show up for a moment before the PWM signal is reconfigured to 50% duty cycle at the start of `test_pwm_nsec`